cmd/dep/init.go

```diff
@@ -113,39 +119,26 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	sm.UseDefaultSignalHandling()
 	defer sm.Release()
 
-	pd, err := getProjectData(ctx, pkgT, cpr, sm)
-	if err != nil {
-		return err
-	}
-	m := &dep.Manifest{
-		Dependencies: pd.constraints,
-	}
-
-	// Make an initial lock from what knowledge we've collected about the
-	// versions on disk
-	l := &dep.Lock{
-		P: make([]gps.LockedProject, 0, len(pd.ondisk)),
+	var rootAnalyzer rootProjectAnalyzer
+	var analyzer gps.ProjectAnalyzer
+	if cmd.skipTools {
+		rootAnalyzer = newGopathAnalyzer(ctx.Loggers, ctx, pkgT, cpr, sm)
+		analyzer = dep.Analyzer{}
+	} else {
+		rootAnalyzer = compositeAnalyzer{
+			Analyzers: []rootProjectAnalyzer{
+				newGopathAnalyzer(ctx.Loggers, ctx, pkgT, cpr, sm),
+				newImportAnalyzer(ctx.Loggers, sm),
+			}}
```

Comment:
The way I have things organized as of right now:

* `dep/analyzer.go` handles identifying if a project uses dep and importing its config.
* `dep/cmd/dep/importAnalyzer.go` handles organizing all the importers for other tools together and importing config for them recursively for all dependencies. The first tool which identities that its config is present is used, and the remaining tools are not used.
* `dep/cmd/dep/gopathAnalyzer.go` contains all the existing logic for generating an initial manifest and lock based on what is on disk under GOPATH.
* `dep/cmd/dep/compositeAnalyzer.go` lets us combine multiple analyzers, where the results of each is overlaid on top of the results of the previous, combining the results of an importer with whatever is found in the GOPATH.

The init uses either just the `gopathAnalyzer` when `-skip-tools` is present (which is the behavior in master) for the root project and dependencies, or makes a `compositeAnalyzer` to determine the root manifest (first inspecting the GOPATH, then overlaying the import results) and uses the `importAnalyzer` for any dependencies.

Things are still in-flux as I've been developing in small steps. 😀 I haven't decided yet if it makes sense to combine the `gopathAnalazyer` with `compositeAnalyzer`, because they are only used together, but I kind of prefer to not have a hodge-podge struct that does all the things.

Let me know if that makes sense or not.